spec/02-integration/04-admin_api/06-certificates_routes_spec.lua

```diff
@@ -19,210 +20,991 @@ describe("Admin API: #" .. kong_config.database, function()
+        assert.same("bar_cert", json.cert)
+        assert.same("bar_key", json.key)
+
+        -- make sure the certificate got updated in DB
+        res = assert(client:send {
+          method = "GET",
+          path = "/certificates/" .. cert_bar.id,
+        })
+
+        body = assert.res_status(200, res)
+        json = cjson.decode(body)
+        assert.equal("bar_cert", json.cert)
+        assert.equal("bar_key", json.key)
+
+        -- make sure number of snis don't change
+        res = assert(client:send {
+          method  = "GET",
+          path    = "/snis",
+        })
+
+        body = assert.res_status(200, res)
+        json = cjson.decode(body)
+        assert.equal(2, #json.data)
+        assert.equal(2, json.total)
+
+        -- make sure we did not add any certificate
+        res = assert(client:send {
+          method = "GET",
+          path = "/certificates",
+        })
+
+        body = assert.res_status(200, res)
+        json = cjson.decode(body)
+        assert.equal(2, json.total)
+        assert.equal(2, #json.data)
+      end)
+
+      it("returns a conflict when duplicates snis are present in the request", function()
+        local res = assert(client:send {
+          method  = "PUT",
+          path    = "/certificates",
+          body    = {
+            id = cert_bar.id,
+            snis  = "baz.com,baz.com",
+          },
+          headers = { ["Content-Type"] = "application/x-www-form-urlencoded" },
+        })
+
+        local body = assert.res_status(409, res)
+        local json = cjson.decode(body)
+
+        assert.equals("duplicate requested sni name baz.com", json.message)
+
+        -- make sure number of snis don't change
+        res = assert(client:send {
+          method  = "GET",
+          path    = "/snis",
+        })
+
+        body = assert.res_status(200, res)
+        json = cjson.decode(body)
+        assert.equal(2, #json.data)
+        assert.equal(2, json.total)
+
+        -- make sure we did not add any certificate
+        res = assert(client:send {
+          method = "GET",
+          path = "/certificates",
+        })
+
+        body = assert.res_status(200, res)
+        json = cjson.decode(body)
+        assert.equal(2, json.total)
+        assert.equal(2, #json.data)
+      end)
+
+      it("returns a conflict when a pre-existing sni present in " .. 
+        "the request is associated with another certificate", function()
+        local res = assert(client:send {
+          method  = "PUT",
+          path    = "/certificates",
+          body    = {
+            id = cert_bar.id,
+            snis  = "foo.com,baz.com",
+          },
+          headers = { ["Content-Type"] = "application/x-www-form-urlencoded" },
+        })
+
+        local body = assert.res_status(409, res)
+        local json = cjson.decode(body)
+
+        assert.equals("certificate with id " .. cert_foo.id .. 
+          " in use for entry with name foo.com", json.message)
```

Comment:
style: ideally, those extra argument after a line jump would align with the first argument of the caller:
```lua
assert.equals("certificate with id " .. cert_foo.id .. 
              " in use for entry with name foo.com", json.message)
```